README.md

```diff
@@ -418,6 +418,44 @@ func isError(error: Error?) -> Bool {
 }
 ```
 
+### Retain cycle ###
+
+To avoid retain cycle in Swift -- meaning when two objects both have strong references to each other -- use **weak** and **unowned** on your references to avoid having a strong reference on both sides.
+
+* A **weak** reference is a pointer to an object that **doesn't protect** the object from being deallocated by [ARC](https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/AutomaticReferenceCounting.html). The object **can** be nil.
+* A **unowned** reference is a pointer to an object that **doesn't protect** the object from being deallocated by [ARC](https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/AutomaticReferenceCounting.html). The object **can't** be nil.
+
+According to Apple's [documentation](https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/AutomaticReferenceCounting.html):
+
+```Use a weak reference whenever it is valid for that reference to become nil at some point during its lifetime. Conversely, use an unowned reference when you know that the reference will never be nil once it has been set during initialization.```
+
+##### Example #####
+```
+internal final class myClass {
+    weak var delegate: MyDelegate?
+
+    func myFunction() {
+        delegate?.doSomething()
+    }
+    
+    func myFunctionWithClosure() {
+    	let closure = { [weak self, unowned myClassInstance] in
```

Comment:
just a dummy class instance to demonstrate that here we want to use unowned since this class will not be nil